.github/workflows/kafka_demo_test.yml

```diff
@@ -52,54 +57,77 @@ jobs:
         run: | 
           echo "Extracting Environment Variables"
           echo "Triggering event is ${{ github.event_name }}"
+          # uses default SKR but builds new Kafka images from source for PR
           if [[ "${{ github.event_name }}" == "pull_request" ]]; then
             echo "key-release-image=mcr.microsoft.com/aci/skr:2.7" >> $GITHUB_OUTPUT
-            echo "consumer-image=mcr.microsoft.com/acc/samples/kafka/consumer:2.0" >> $GITHUB_OUTPUT
-            echo "producer-image=mcr.microsoft.com/acc/samples/kafka/producer:2.0" >> $GITHUB_OUTPUT
+            echo "consumer-image=private/${{ github.actor }}/acc/samples/kafka/consumer:${{ github.sha }}" >> $GITHUB_OUTPUT
+            echo "producer-image=private/${{ github.actor }}/acc/samples/kafka/producer:${{ github.sha }}" >> $GITHUB_OUTPUT
           else
             echo "key-release-image=${{ format('{0}/{1}', inputs.sidecar-registry, inputs.key-release-image) }}" >> $GITHUB_OUTPUT
             echo "consumer-image=${{ inputs.consumer-image }}" >> $GITHUB_OUTPUT
             echo "producer-image=${{ inputs.producer-image }}" >> $GITHUB_OUTPUT
           fi
 
+  push_images:
+    name: Push Images to Private Registry for Testing
+    needs: [prepare]
+    uses: ./.github/workflows/_push_image.yml
+    if: github.event_name == 'pull_request'
+    secrets: inherit
+    strategy:
+      matrix:
+        image: ["kafka/consumer", "kafka/producer"]
+    with:
+      image_tag: "private/${{ github.actor }}/acc/samples/${{ matrix.image }}:${{ github.sha }}"
+      docker_context: ${{ matrix.image }}
+      repo_type: private
+
   create-aks-cluster:
     name: Create AKS Cluster
-    needs: [prepare]
+    needs: [prepare, push_images]
     uses: ./.github/workflows/_create_aks_cluster.yml
+    if: always() && (needs.push_images.result == 'success' || needs.push_images.result == 'skipped')
```

Comment:
Just curious, is the `always()` needed here? Would the conditional be equal to
`if: (needs.push_images.result == 'success' || needs.push_images.result == 'skipped')`

Same question about the conditional in `setup-aks-cluster`.

Reply:
This stack overflow explains why the always() is included: https://stackoverflow.com/questions/66343052/github-actions-or-operator-for-needs-clause. Essentially, always is needed to make sure this job doesn't get skipped when we skip the push-images job.